microceph/ceph/services.go

```diff
@@ -16,11 +15,22 @@ import (
 	"github.com/canonical/microcluster/state"
 
 	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/common"
 	"github.com/canonical/microceph/microceph/database"
 	"github.com/tidwall/gjson"
 )
 
-type Set map[string]struct{}
+type Set map[string]interface{}
+
+func (sub Set) Keys() []string {
+	keys := make([]string, len(sub))
+
+	for key := range sub {
+		keys = append(keys, key)
```

Comment:
why not `keys[i] = key` (also changing the `for` loop to use an index variable)? It would be symmetrical with other calls.

Reply:
ack